maketgz

```diff
@@ -192,32 +219,38 @@ gzip -dc $targz | bzip2 --best > $bzip2
 
 xz="curl-$version.tar.xz"
 echo "Generating $xz"
-gzip -dc $targz | xz -6e - > $xz
+gzip -dc "$targz" | xz -6e - > "$xz"
 
 ############################################################################
 #
 # Now make a zip archive from the tar.gz original
 #
 makezip() {
-  rm -rf $tempdir
-  mkdir $tempdir
-  cd $tempdir
-  gzip -dc ../$targz | tar -xf -
-  find . | zip $zip -@ >/dev/null
-  mv $zip ../
+  rm -rf "$tempdir"
+  mkdir "$tempdir"
+  cd "$tempdir"
+  gzip -dc "../$targz" | tar -xf -
+  find . -depth -type d -exec touch -c -t "$filestamp" '{}' +
+  find . | sort | zip -9 -X "$zip" -@ >/dev/null
+  mv "$zip" ../
   cd ..
-  rm -rf $tempdir
+  rm -rf "$tempdir"
 }
 
 zip="curl-$version.zip"
 echo "Generating $zip"
 tempdir=".builddir"
 makezip
 
+# Set deterministic timestamp
+touch -c -t "$filestamp" "$targz" "$bzip2" "$xz" "$zip"
+
 echo "------------------"
 echo "maketgz report:"
 echo ""
-ls -l $targz $bzip2 $zip $xz
+ls -l "$targz" "$bzip2" "$xz" "$zip"
+
+command -v openssl >/dev/null 2>&1 && openssl dgst -sha256 "$targz" "$bzip2" "$xz" "$zip"
```

Comment:
This adds an unfortunate dependency on the openssl tool which I think we can avoid. How about just just use `sha256sum`?

Reply:
I choose openssl because some form of it is available in most envs (tests/certs/scripts/*.sh already depend on it). `sha256sum` is missing by default from macOS, also possibly from BSDs.

It should be easy to install on any platform though, so why not.